sn_interface/src/messaging/signature_aggregator.rs

```diff
@@ -71,90 +67,30 @@ impl SignatureAggregator {
 
-        self.map
-            .entry(hash)
-            .or_insert_with(State::new)
-            .add(sig_share)
-            .map(|signature| SectionSig {
-                public_key,
-                signature,
-            })
-    }
-
-    fn remove_expired(&mut self) {
-        let expiration = self.expiration;
-        let mut to_remove = vec![];
-
-        for (digest, state) in &self.map {
-            if state.modified.elapsed() >= expiration {
-                to_remove.push(*digest);
-            }
-        }
-
-        self.map.retain(|digest, _| !to_remove.contains(digest))
-    }
-}
+        // save the sig share
+        let current_shares = self.map.entry(hash).or_default();
+        let _ = current_shares.insert(sig_share.clone());
 
-impl Default for SignatureAggregator {
-    fn default() -> Self {
-        Self::with_expiration(DEFAULT_EXPIRATION)
-    }
-}
-
-/// Error returned from `SignatureAggregator::add`.
-#[derive(Debug, Error)]
-pub enum Error {
-    /// There are not enough signature shares yet, more need to be added. This is not a failure.
-    #[error("not enough signature shares")]
-    NotEnoughShares,
-    /// The signature share being added is invalid. Such share is rejected but the already collected
-    /// shares are kept intact. If enough new valid shares are collected afterwards, the
-    /// aggregation might still succeed.
-    #[error("signature share is invalid")]
-    InvalidShare,
-    /// The signature combination failed even though there are enough valid signature shares. This
-    /// should probably never happen.
-    #[error("failed to combine signature shares: {0}")]
-    Combine(#[from] bls::error::Error),
-}
-
-#[derive(Debug)]
-struct State {
-    shares: BTreeMap<usize, bls::SignatureShare>,
-    modified: Instant,
-}
-
-impl State {
-    fn new() -> Self {
-        Self {
-            shares: Default::default(),
-            modified: Instant::now(),
-        }
-    }
-
-    fn add(&mut self, sig_share: SectionSigShare) -> Result<bls::Signature, Error> {
-        if self
-            .shares
-            .insert(sig_share.index, sig_share.signature_share)
-            .is_none()
-        {
-            self.modified = Instant::now();
-        } else {
-            // Duplicate share
-            return Err(Error::NotEnoughShares);
-        }
-
-        if self.shares.len() > sig_share.public_key_set.threshold() {
+        // try aggregate
+        if current_shares.len() > sig_share.public_key_set.threshold() {
             let signature = sig_share
                 .public_key_set
-                .combine_signatures(&self.shares)
-                .map_err(Error::Combine)?;
-
-            self.shares.clear();
+                .combine_signatures(
+                    current_shares
+                        .iter()
+                        .map(|s| (s.index, s.signature_share.clone())),
+                )
+                .map_err(AggregatorError::FailedToCombineSigShares)?;
```

Comment:
We don't need to map the error here explicitly since you have `#[from] bls::error::Error` in `FailedToCombineSigShares(#[from] bls::error::Error)`.

`?` will do the conversion automatically.
```suggestion
                .combine_signatures(
                    current_shares
                        .iter()
                        .map(|s| (s.index, s.signature_share.clone())),
                )?;
```